metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/collect/CollectStrategy.java

```diff
@@ -114,6 +115,17 @@ private void collect(final Connector currentConnector, final String hostname) {
 			return;
 		}
 
+		// Run PreSourceStrategy that executes pre sources
+
+		final PreSourceStrategy preSourceStrategy = PreSourceStrategy
```

Comment:
preSourceStrategy needs to be instantiated with `currentConnector`. Then it should  process the `currentConnector`'s pre sources only.